src/mediapipe/imageSegmenter/audioNativeRenderer/MatrixAudioRenderer.tsx

```diff
@@ -0,0 +1,215 @@
+      `track will not get rendered`,
+    );
+  };
+
+  const tracks = useTracks(
+    [
+      Track.Source.Microphone,
+      Track.Source.ScreenShareAudio,
+      Track.Source.Unknown,
+    ],
+    {
+      updateOnlyOn: [],
+      onlySubscribed: true,
+    },
+  ).filter((ref) => {
+    const isValid = validIdentities?.includes(ref.participant.identity);
+    if (!isValid) logInvalid(ref.participant.identity, validIdentities);
+    return (
+      !ref.participant.isLocal &&
+      ref.publication.kind === Track.Kind.Audio &&
+      isValid
+    );
+  });
+
+  const { pan, volume } = useEarpiceAudioConfig();
+
+  return (
+    <div style={{ display: "none" }}>
+      {tracks.map((trackRef) => (
+        <StereoPanAudioTrack
+          key={getTrackReferenceId(trackRef)}
+          trackRef={trackRef}
+          muted={muted}
+          stereoPan={pan}
+          volumeFactor={volume}
+        />
+      ))}
+    </div>
+  );
+}
+
+interface StereoPanAudioTrackProps {
+  /**
+   * This additional volume factor is applied to all audio tracks from livekit.
+   *
+   * @remark It does not interfere with the `setVolume` of the track itself.
+   * The seperta `volume` property will override/conflict with what is set using `track.setVolume`
+   */
+  volumeFactor: number;
+  /**
+   * The left right stereo pan of the audio tracks.
+   * This is a value between -1 and 1, where -1 is full left, 0 is center, and 1 is full right.
+   */
+  stereoPan: number;
+  muted?: boolean;
+}
+
+function StereoPanAudioTrack({
+  trackRef,
+  stereoPan,
+  volumeFactor,
+  muted,
+  ...props
+}: StereoPanAudioTrackProps &
+  AudioTrackProps &
+  React.RefAttributes<HTMLAudioElement>): ReactNode {
+  // This component is used to mimic earpice audio on iphones.
+  // The safari audio devices do not expose an earpice audio device.
+  // This is achieved by using the audioContext pan node to only use one of the stereo channels.
+
+  // This component does get additionally complicated because of a safari bug.
+  // (see: https://bugs.webkit.org/show_bug.cgi?id=251532
+  // and the related issues: https://bugs.webkit.org/show_bug.cgi?id=237878
+  // and https://bugs.webkit.org/show_bug.cgi?id=231105)
+  //
+  // AudioContext gets stopped if the webview gets moved into the background.
+  // Once the phone is in standby audio playback will stop.
+  // So we can only use the pan trick in the following conditions
+  // - webview is visible
+  // - stereoPan is not 0
+  //
+  // The architecture of the component is as follows:
+  // the `audioContext` state is used to determine if we should use the audio context or not.
+  // the actual `audioContext` is stored in `audioContextRef` to minimize reallocations.
+  // whenever we notice the audioContext is active but should not: `!!audioContext !== shouldUseAudioContext`
+  // we remount the audio tracks using a true -> false -> true transition for the `trackReady` state.
+  const audioContextRef = useRef(new AudioContext());
```

Comment:
This will create a new AudioContext on each render, which is dangerous: you need to at least call the `close()` method.

```suggestion
  const audioContext = useMemo(() => new AudioContext(), []);
  useEffect(() => () => audioContext.close(), [audioContext]);
```

But in general we've seen this cause big resource leaks in the past, back when walkie-talkie mode was a thing. Is there some way that we could just create a single AudioContext in the global scope and use that across the entire app? (that is, use the same context in `useAudioContext` too?) Given that an accumulation of AudioContexts can cause stuttering audio, I think we should consider them expensive.

Reply:
Are you sure it would create one each render.
Or do you mean a new one on each creation of the compoenent?

But that will be the same for the memo version.

I am not sure we can have the same context with different destinations?
Now they are all just connected to one html element.
Lk also provides a way to pass a single mixed audio context to the lk room.
But this would be an entirely differnt architecture and I am not sure we would be able to enforece the correct remounting ourder as easily when switching from audio context to not using audio context.

I would postpone a central audioContext refactor once we have a decision if locking ios should mute audio or switch back to native html audio?